### PR TITLE
ci: pin exact Java version

### PR DESCRIPTION
**Pull request by jjohannes**
**Description**:

This is important for deterministic and reproducible builds. Without this, the version automatically changes when there is a new release.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
         uses: actions/setup-java@8df1039502a15bceb9433410b1a100fbe190c53b # v4.5.0
         with:
           distribution: temurin
-          java-version: 17
+          java-version: '17.0.13'
```
Comment (rbarkerSL) [inline .github/workflows/build.yml:46]: Is there a reason we are on 17.0.13 instead of 21.0.4?

**Review by rbarkerSL — COMMENTED** [review]
Curious about this java version, should we not be building against the latest?

File: .github/workflows/build.yml (hunk 1 above)
Comment (jjohannes) [reply .github/workflows/build.yml:46]: For moving from 17 to 21, there is a separate issue: 
https://github.com/hashgraph/hedera-sdk-java/issues/2075

One concern may be to not force users of the SDK to update their projects (yet).

Comment (0xivanov) [reply .github/workflows/build.yml:46]: @jjohannes @rbarkerSL Updating to java 21 might break some users and we haven't gotten to it yet, so the runners should use the same version as the target for the project - 17

**Review by rbarkerSL — APPROVED** [review]

**Pull request merged**